File: pathfinder-frontend/src/model/character/Size.ts
class Size {
  public static readonly NONE = new Size(0, '-', 'N/A', 0);
  public static readonly FINE = new Size(1, 'F', 'Fine', 8);
  public static readonly DIMINUTIVE = new Size(2, 'D', 'Diminutive', 4);
  public static readonly TINY = new Size(3, 'T', 'Tiny', 2);
  public static readonly SMALL = new Size(4, 'S', 'Small', 1);
  public static readonly MEDIUM = new Size(5, 'M', 'Medium', 0);
  public static readonly LARGE = new Size(6, 'L', 'Large', -1);
  public static readonly HUGE = new Size(7, 'H', 'Huge', -2);
  public static readonly GARGANTUAN = new Size(8, 'G', 'Gargantuan', -4);
  public static readonly COLOSSAL = new Size(9, 'C', 'Colossal', -8);

  public static fromId(id: number): Size | undefined {
    switch (id) {
      case 0: return this.NONE;
      case 1: return this.FINE;
      case 2: return this.DIMINUTIVE;
      case 3: return this.TINY;
      case 4: return this.SMALL;
      case 5: return this.MEDIUM;
      case 6: return this.LARGE;
      case 7: return this.HUGE;
      case 8: return this.GARGANTUAN;
      case 9: return this.COLOSSAL;
    }
    return undefined;
  }

  private constructor(public readonly id: number,
                      public readonly shortName: string,
                      public readonly longName: string,
                      public readonly acModifier: number) {
  }
}

export default Size;

File: pathfinder-generator/src/main/java/pathfinder/model/pathfinder/Size.java
package pathfinder.model.pathfinder;

import java.util.Optional;
import lombok.RequiredArgsConstructor;
import pathfinder.model.Id;

@RequiredArgsConstructor
public enum Size {
    NONE(Id.of("size:n"), "N", "N/A", 0, 0),
    FINE(Id.of("size:f"), "F", "Fine", 1, 8),
    DIMINUTIVE(Id.of("size:d"), "D", "Diminutive", 2, 4),
    TINY(Id.of("size:t"), "T", "Tiny", 3, 2),
    SMALL(Id.of("size:s"), "S", "Small", 4, 1),
    MEDIUM(Id.of("size:m"), "M", "Medium", 5, 0),
    LARGE(Id.of("size:l"), "L", "Large", 6, -1),
    HUGE(Id.of("size:h"), "H", "Huge", 7, -2),
    GARGANTUAN(Id.of("size:g"), "G", "Gargantuan", 8, -4),
    COLOSSAL(Id.of("size:c"), "C", "Colossal", 9, -8);

    public static Optional<Size> findByLongName(String longName) {
        for (Size size : Size.values()) {
            if (size.longName().equalsIgnoreCase(longName)) {
                return Optional.of(size);
            }
        }
        return Optional.empty();
    }

    private final Id id;
    private final String shortName;
    private final String longName;
    private final int number;
    private final int sizeMod;

    public Id id() {
        return id;
    }

    public String shortName() {
        return shortName;
    }

    public String longName() {
        return longName;
    }

    public int number() {
        return number;
    }

    public int sizeMod() {
        return sizeMod;
    }
}
